dashboard/src/pages/apps/new.tsx

```diff
@@ -187,161 +203,178 @@ const InputFormRadio = styled('div', {
     gap: '12px',
   },
 })
-
-const RepositoriesContainer = styled('div', {
+styled('div', {
   base: {
     display: 'flex',
     flexDirection: 'column',
     gap: 'px',
   },
 })
-
 const SettingsContainer = styled('div', {
   base: {
     display: 'flex',
     flexDirection: 'column',
     gap: '10px',
   },
 })
-
-interface Website {
-  fqdn: string
-  path_prefix: string
-  strip_prefix: boolean
-  https: boolean
-  h2c: boolean
-  http_port: number
-  authenticationType: number
-}
-
-const EmptyWebsite: Website = {
-  fqdn: '',
-  path_prefix: '',
-  strip_prefix: false,
-  https: false,
-  h2c: false,
-  http_port: 0,
-  authenticationType: AuthenticationType.OFF,
-}
-
 interface WebsiteProps {
-  website: Website
-  setWebsite: (Website) => void
+  website: CreateWebsiteRequest
+  setWebsite: (valueName, value) => void
```

Comment:
[must] これlintエラー出ないんですね...

![image](https://github.com/traPtitech/NeoShowcase/assets/18257693/7890929b-d96f-46fc-92a7-7fa25ce131d8)

setWebsiteの型を見ると、`(valueName: any, value: any) => void` となっていると思います。型を指定していない & 推測できないので、何も型が付いていない状態です。
型がanyになる状況は極力避けたいので、適切な型を指定するようにしてみてください

この場合だとstoreのsetterに関係する関数だと思うので、そこを参照するとよいでしょう（とは言ってもSetStoreFunctionの型定義は見にくいことこの上無いんですが...）
例えば今回だったら1階層だけのsetが必要なので、↓みたいに書いてはどうでしょうか
これにした上で、例えば `setWebsite('https', 'test')` などとすると `TS2345: Argument of type 'string' is not assignable to parameter of type 'boolean'.` と出て、型エラーに気づくことができます。
```suggestion
  setWebsite: <T extends keyof CreateWebsiteRequest>(valueName: T, value: CreateWebsiteRequest[T]) => void
```

Reply:
もしくは（型がきれいになるカリー化が思いつかないので）
```suggestion
  setWebsite: SetStoreFunction<CreateWebsiteRequest>
```
として、下のように渡してやるとか... さすがに気持ち悪いので上のコメントの方がいいと思うけど参考程度に
```
setWebsite={(...args) => setWebsiteConfigs.apply(this, [i(), ...args])}
```

Reply:
凄くなるほどです、これに関しては仕方がないのかな～って思ってましたが、このような賢い書き方ができるんですね。